Unity/opentok/holo/holo/Capturer.mm

```diff
@@ -0,0 +1,373 @@
+#import "Capturer.h"
+
+#include <memory>
+
+#import <sdk/objc/base/RTCVideoFrameBuffer.h>
+#import <sdk/objc/components/video_frame_buffer/RTCCVPixelBuffer.h>
+#import <sdk/objc/vonage/capturer/VonageRTCCameraVideoCapturer.h>
+
+#import <AugmentedCompression.h>
+
+#define COMPRESSED_SIZE 320 * 1024
+
+typedef NS_ENUM(int32_t, OTCapturerErrorCode) {
+
+    OTCapturerSuccess = 0,
+
+    /** Publisher couldn't access to the camera */
+    OTCapturerError = 1650,
+
+    /** Publisher's capturer is not capturing frames */
+    OTCapturerNoFramesCaptured = 1660,
+
+    /** Publisher's capturer authorization failed */
+    OTCapturerAuthorizationDenied = 1670,
+};
+
+@interface DepthDataCompressor : NSObject<RTC_OBJC_TYPE(RTCVideoDepthDataDelegate)>
+
+@end
+
+@implementation DepthDataCompressor
+
+- (BOOL)compressInputArray:(const std::unique_ptr<uint8_t[]> &)inputArray inputSize:(uint32_t)inputSize outputArray:(std::unique_ptr<uint8_t[]> &)outputArray outputSize:(uint32_t &)outputSize {
+    return Holographic::Compression::compress(inputArray, inputSize, outputArray, outputSize);
+}
+
+- (BOOL)compressInputArray:(nonnull CVPixelBufferRef)depthDataMap outputArray:(std::unique_ptr<uint8_t[]> &)outputArray outputSize:(uint32_t &)outputSize {
+    return Holographic::Compression::compress(depthDataMap, outputArray, outputSize);
+}
+
+@end
+
+@interface Capturer()
+@property (nonatomic) RTC_OBJC_TYPE(VonageRTCCameraVideoCapturer) *capturer;
+@property (nonatomic) DepthDataCompressor *depthDataCompressor;
+
+- (void) setupListenerBlocks;
+@end
+
+@implementation Capturer {
+    __weak id<OTVideoCaptureConsumer> _videoCaptureConsumer;
+    OTVideoFrame* _videoFrame;
+
+    uint32_t _captureWidth;
+    uint32_t _captureHeight;
+    NSString* _capturePreset;
+
+    BOOL _capturing;
+
+    enum OTCapturerErrorCode _captureErrorCode;
+}
+
+@synthesize videoCaptureConsumer = _videoCaptureConsumer;
+@synthesize videoContentHint;
+
+-(id)init {
+    self = [super init];
+    if (self) {
+        _capturePreset = AVCaptureSessionPreset640x480;
+        [[self class] dimensionsForCapturePreset:_capturePreset
+                                           width:&_captureWidth
+                                          height:&_captureHeight];
+        _capture_queue = dispatch_queue_create("com.vonage.holo.Capturer",
+                                               DISPATCH_QUEUE_SERIAL);
+        _videoFrame = [[OTVideoFrame alloc] initWithFormat:
+                       [OTVideoFormat videoFormatNV12WithWidth:_captureWidth
+                                                        height:_captureHeight]];
+    }
+    return self;
+}
+
+-(id)initWithCapturePreset: (NSString*)preset {
```

Comment:
should be `AVCaptureSessionPreset` and not `NSString`

Reply:
It is an alias (https://developer.apple.com/documentation/avfoundation/avcapturesessionpreset). I can change it.